### PR TITLE
Ramda: Make partial and partialRight binary fns

### PR DESCRIPTION
**Pull request by angeloocana**
Change Added to Ramda 0.18.0

Pull Request: https://github.com/ramda/ramda/pull/1433
Upgrade guide: https://github.com/ramda/ramda/issues/1436

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/ramda/ramda/pull/1433
- [ ] Increase the version number in the header if appropriate.

### CODE REVIEW

File: types/ramda/index.d.ts
Comment (samsonkeung) [inline types/ramda/index.d.ts]: This should be removed because the end result should have the same type as `fn`'s return type instead of `any`

File: types/ramda/index.d.ts
Comment (samsonkeung) [inline types/ramda/index.d.ts]: Same as above

**Review by samsonkeung — CHANGES_REQUESTED** [review]
See comments on changes

**Comment by RyanCavanaugh** [comment]
@angeloocana Please address comments from the code reviewers.

**Comment by RyanCavanaugh** [comment]
@samsonkeung - Thanks for your review of this PR! Can you please look at the new code and update your review status if appropriate?

**Review by samsonkeung — APPROVED** [review]
Thank you!

**Comment by RyanCavanaugh** [comment]
Approved by a listed owner. PR ready to merge pending express review by a maintainer.

**Pull request merged**
